src/utils/utils.ts

```diff
@@ -101,17 +114,44 @@ export default class Utilities {
             "purpose"
         ]
 
-        satellitesCreation.forEach((value) => {
-            if (Object.entries(value).some(propertyAndValue => requiredFields.includes(propertyAndValue[0]) && propertyAndValue[1] === '')) {
-                return incompleteSatellites.push(value)
+        const satellitesWithRequiredFields = satellitesFromFile.filter((item) => {
+            const currentKeys = Object.keys(item)
+            return requiredFields.every(item => currentKeys.includes(item))
+        })
+
+        const standardizedSatellites = this.standardizeSatellitesData(satellitesWithRequiredFields)
+
+        const numFromStr = coerce(number(), string(), (value) => parseFloat(value))
+
+        const validatedSatellites = standardizedSatellites.filter(element => {
+            try {
+                create(element.longitudeOfGeo, numFromStr)
+                create(element.perigee, numFromStr)
+                create(element.apogee, numFromStr)
+                create(element.inclination, numFromStr)
+                create(element.period, numFromStr)
+                create(element.launchMass, numFromStr)
+                create(element.dryMass, numFromStr)
+                create(element.power, numFromStr)
+                create(element.expectedLifetime, numFromStr)
+                create(element.cospar, numFromStr)
+                create(element.norad, numFromStr)
+            } catch {
+                return false
             }
-            completeSatellites.push(value)
+            return true
         })
 
-        return {
-            completeSatellites,
-            incompleteSatellites
-        }
+        return validatedSatellites
+    }
+
+    static prePopulateDatabase(satellites: SatelliteType[]): Promise<SatelliteType[]> {
+        const satellitesCreation: Promise<SatelliteType>[] = []
+        satellites.forEach((satellite) => {
```

Comment:
You could just use .map and map satellities to their creation promises. No need to use for each. You in general use for each a lot which is basically a bit cleaner loop.